pkgs/applications/audio/projectm/default.nix

```diff
@@ -1,57 +1,66 @@
-{ stdenv, fetchurl, fetchpatch, pkgconfig, cmake
-, glew, ftgl, ttf_bitstream_vera
-, withQt ? true, qt4
-, withLibvisual ? false, libvisual, SDL
-, withJack ? false, libjack2
-, withPulseAudio ? true, libpulseaudio
-}:
+{ mkDerivation
+, lib
+, fetchFromGitHub
 
-assert withJack       -> withQt;
-assert withPulseAudio -> withQt;
+, autoreconfHook
+, pkgconfig
+, SDL2
+, qtdeclarative
+, libpulseaudio
+, glm
+, which
+}:
 
-stdenv.mkDerivation {
-  name = "projectm-2.1.0";
+mkDerivation rec {
+  pname = "projectm";
+  version = "3.1.3";
 
-  meta = {
-    description = "Music Visualizer";
-    homepage = http://projectm.sourceforge.net/;
-    license = stdenv.lib.licenses.lgpl21Plus;
-    platforms = stdenv.lib.platforms.linux;
+  src = fetchFromGitHub {
+    owner = "projectM-visualizer";
+    repo = "projectM";
+    rev = "v${version}";
+    sha256 = "1mjnahr694phksmvc069y89rv85s4l2z9fixkc3l1f5qj2vgn4sy";
   };
 
-  src = fetchurl {
-    url = "mirror://sourceforge/projectm/2.1.0/projectM-complete-2.1.0-Source.tar.gz";
-    sha256 = "1vh6jk68a0jdb6qwppb6f8cbgmhnv2ba3bcavzfd6sq06gq08cji";
-  };
+  nativeBuildInputs = [
+    pkgconfig
+    autoreconfHook
+    which
+  ];
 
-  patch_gcc6 = fetchpatch {
-    url = https://raw.githubusercontent.com/gentoo/gentoo/45abd63abc6644b6e177c057b5b42d894dbf8e29/media-libs/libprojectm/files/libprojectm-2.1.0-fix-c++14.patch;
-    sha256 = "1i50scxv1jlqvb3jm3sql89a7wqckxhlpvnhz20vvmm1kii6lrsn";
-  };
+  outputs = [ "bin" "dev" "out" ]; # NOTE: 2019-10-05: libprojectM in "lib" creates a cyclic dependency on the "out"
```

Comment:
I suggest you don't use multiple outputs. The gains are very small because of a single one reverse dependency (which you break because of this). And you get extra maintenance burden because of a custom autotools/makefile-based build system.

Reply:
done